samples/code-samples/Queries/AsyncExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents.Linq;
+
+namespace DocumentDB.Samples.Queries
+{
+    public static class AsyncExtensions
+    {
+        public static async Task<IEnumerable<T>> QueryAsync<T>(this IQueryable<T> query)
+        {
+            var documentQuery = query.AsDocumentQuery();
+            var batches = new List<List<T>>();
+
+            do
+            {
+                batches.Add((await documentQuery.ExecuteNextAsync<T>()).ToList());
```

Comment:
How about using defining `var batches = new List<T>();` and using `.AddRange()` instead of `.Add()` - that will use less memory and save you from having to `.SelectMany(x => x)`.